### PR TITLE
[RFC] doc: Add transition instructions to nvim_from_vim.txt.

### PR DESCRIPTION
**Pull request by fwalch**
Ties in with #1885, because that will mention `:h nvim-intro` on the startup screen.

Ping @Pyrohh.

### CODE REVIEW

File: runtime/doc/nvim_intro.txt [hunk 1]
```diff
@@ -4,14 +4,60 @@
...
+    ln -s ~/.vimrc ~/.nvimrc
+    ln -s ~/.vim ~/.nvim
+<
+See |nvim-topics|, especially |nvim-python| and |nvim-clipboard|, for
+additional software you might need to install to use all of Neovim's features.
+
+Your Vim configuration might not be entirely compatible with Neovim. For a
+full list of differences between Vim and Neovim, see |vim-differences|.
+
+The `ttymouse` option, for example, was removed from Neovim (mouse support
```
Comment (Pyrohh) [inline runtime/doc/nvim_intro.txt]: Should probably be:
```
|'ttymouse'|
```

File: runtime/doc/nvim_intro.txt [hunk 2]
```diff
@@ -4,14 +4,60 @@
...
+should work without it). If you use the same `vimrc` for Vim and Neovim,
+consider guarding `ttymouse` in your configuration like so:
```
Comment (Pyrohh) [inline runtime/doc/nvim_intro.txt]: same here

File: runtime/doc/nvim_intro.txt [hunk 3]
```diff
@@ -4,14 +4,60 @@
...
+    endif
+<
+Conversely, if you have Neovim specific configuration items, you could do
+this:
+>
+    if has('nvim')
+        tnoremap <Esc> <C-\><C-n>
+    endif
+<
+For a more granular approach, use |:exists()|:
```
Comment (Pyrohh) [inline runtime/doc/nvim_intro.txt]: Should probably be `|exists()|` (no colon). The actual tag lacks a colon and the colon indicates it's being done in command mode, which it isn't in this case.

File: runtime/doc/nvim_intro.txt [hunk 4]
```diff
@@ -4,14 +4,60 @@
...
+    ln -s ~/.vim ~/.nvim
+<
+See |nvim-topics|, especially |nvim-python| and |nvim-clipboard|, for
+additional software you might need to install to use all of Neovim's features.
+
+Your Vim configuration might not be entirely compatible with Neovim. For a
+full list of differences between Vim and Neovim, see |vim-differences|.
+
+The `ttymouse` option, for example, was removed from Neovim (mouse support
+should work without it). If you use the same `vimrc` for Vim and Neovim,
```
Comment (Pyrohh) [inline runtime/doc/nvim_intro.txt]: Should probably link to `|vimrc|`.

**Comment by Pyrohh** [comment]
Now that I think about it, I'm not very fond of placing the index to nvim specific topics at the bottom. This transitioning information is most likely not going to be helpful past the first hour of using Nvim.

I propose we keep the document name of `nvim_intro.txt`, but move the index (section 3) to a different page, perhaps `nvim.txt` or `nvim_index.txt`, then add `nvim_intro.txt` as the first item in the index, or perhaps place it as its own sentence somewhere around:
```
Introduction to Nvim					*nvim* *nvim-intro*

This is an introduction for Vim users who are just getting started with Nvim.
It is not meant for Vim beginners. For a basic introduction to Vim,
see |help.txt|.

<------------ SOMEWHERE HERE ------------------->

For now, it is just an index with the most relevant topics/features that
differentiate Nvim from Vim:

1. Differences from Vim		 |vim-differences|
2. Msgpack-RPC			 |msgpack-rpc|
3. Job control			 |job-control|
4. Python plugins		 |nvim-python|
5. Clipboard integration	 |nvim-clipboard|
6. Remote plugins		 |remote-plugin|
7. Provider infrastructure	 |nvim-provider|
8. Integrated terminal emulator  |nvim-terminal-emulator|
```

**Comment by fwalch** [comment]
@Pyrohh Added an `nvim.txt` with the topic overview, let me know what you think!

**Comment by Pyrohh** [comment]
I think it looks pretty good, although `gmake html` prints some warnings:

```
============================================
nvim_intro.txt, line 27, pointer: >>nvim-topics<<
See YXXYnvim-topicsYXXY, especially YXXYnvim-pythonYXXY and YXXYnvim-clipboardYXXY, for
============================================
nvim_intro.txt, line 33, pointer: >>ttymouse<<
The YXXYttymouseYXXY option, for example, was removed from Neovim (mouse support
============================================
nvim_intro.txt, line 35, pointer: >>ttymouse<<
consider guarding YXXYttymouseYXXY in your configuration like so:
```

The `ttymouse` refs should be `'ttymouse'`, my bad. There doesn't seem to be an `nvim-topics` tag.

Note to self, some links on the Wiki index need to be updated after this.

**Comment by fwalch** [comment]
Sorry @Pyrohh, I forgot to check the tags *again* before committing. Should be fixed now.

**Comment by Pyrohh** [comment]
No big deal :smile:

Anyways, this LGTM. Nice improvement :+1:

**Comment by Pyrohh** [comment]
@fwalch do you mind if I merge this, or are you waiting on more reviews (cc @fmoralesc)?

**Comment by fmoralesc** [comment]
# DON'T LET YOUR DREAMS BE DREAMS, @Pyrohh

**Comment by Pyrohh** [comment]
> DON'T LET YOUR DREAMS BE DREAMS

:+1: Added to 0.1 milestone.

**Comment by fmoralesc** [comment]
> :+1: Added to 0.1 milestone.

It *could* actually be the milestime codename.

**Comment by Pyrohh** [comment]
:grin: 

In all seriousness @fmoralesc, what do you think of this PR?

**Comment by fmoralesc** [comment]
It looked good to me when I saw it yesterday, but now that it look at it again, perhaps there should be a link to the `|nvim|` somewhere. Not everyone who'll be looking at the intro will want to transition from vim (I hope). Perhaps it would be even cleaner to move this file to `nvim_from_vim.txt`, and leave the `nvim.txt` as the canonical "intro"?

**Comment by Pyrohh** [comment]
> perhaps there should be a link to the |nvim| somewhere

The tag corresponding to `|nvim|` is in `nvim.txt`. Do you think it should be somewhere else too?

> Not everyone who'll be looking at the intro will want to transition from vim (I hope).

Yes, that's why I recommended moving the index (`nvim.txt`) to a top level filename.

> Perhaps it would be even cleaner to move this file to nvim_from_vim.txt, and leave the nvim.txt as the canonical "intro"?

:+1: That should help reduce any ambiguity.

**Comment by fmoralesc** [comment]
>  Do you think it should be somewhere else too?

No, but I think it should appear in the transition document, so the link can be followed in a sort of "where do I go from here?" way.

**Comment by fwalch** [comment]
Updated. Is that similar to what you meant?

I've been struggling to come up with a good "where do I go from here" sentence for the end of the "Transitioning from Vim" section, so suggestions are welcome! :-)

**Comment by fmoralesc** [comment]
Yes, that is it precisely. 

What about: "Now you have should be able to explore Neovim more comfortably, check |nvim| for more information"?

**Comment by Pyrohh** [comment]
> Now you have should be able

`DAMN SON`

File: runtime/doc/nvim_from_vim.txt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+Nvim is emphatically a fork of Vim, so compatibility to Vim should be pretty
+good.
+
+To start the transition, link your previous configuration so Nvim can use
+it:
+>
+    ln -s ~/.vimrc ~/.nvimrc
+    ln -s ~/.vim ~/.nvim
+<
+See |nvim|, especially |nvim-python| and |nvim-clipboard|, for additional
```
Comment (Pyrohh) [inline runtime/doc/nvim_from_vim.txt]: now that the tag is back, maybe `|nvim|` -> `|nvim-intro|`?

**Comment by Pyrohh** [comment]
besides the comments I made (and @fmoralesc wording suggestion, which I like), LGTM

**Comment by fmoralesc** [comment]
Sorry for that! Giving advice, even... LOL.

**Comment by Pyrohh** [comment]
> Sorry for that! Giving advice, even... LOL.

Oh lol, I wasn't even being sarcastic. I got your intention :)

**Comment by fwalch** [comment]
Updated :-)

**Comment by justinmk** [comment]
LGTM

**Comment by Pyrohh** [comment]
Merged, thanks everyone.

**Pull request closed without merging**